src/Controller/AdminController.php

```diff
@@ -590,6 +590,41 @@ public function userEditAction($username, Request $request, UserManagerInterface
         ]);
     }
 
+    /**
+     * @Route("/admin/user/{username}/delete", name="admin_user_delete")
+     */
+    public function userDeleteAction($username, Request $request, UserManagerInterface $userManager)
+    {
+        $user = $userManager->findUserByUsername($username);
+
+        if (!$user) {
+            throw $this->createNotFoundException();
+        }
+
+        $anonymousEmail = sprintf('anon%s@coopcycle.org', bin2hex(random_bytes(8)));
+
+        $user->setEmail($anonymousEmail);
+        $user->setEmailCanonical($anonymousEmail);
+        $user->setEnabled(false);
+
+        $customer = $user->getCustomer();
+        if (null !== $customer) {
+            $customer->setEmail($anonymousEmail);
+            $customer->setEmailCanonical($anonymousEmail);
```

Comment:
Can we also clear the firstName / lastName? 

```php
$customer->setFullName('');
```

Reply:
done :)